tests/upgrade_path/test_upgrade_path.py

```diff
@@ -238,14 +238,18 @@ def test_upgrade_path(localhost, duthosts, rand_one_dut_hostname, ptfhost, upgra
                 # advance-reboot test (on ptf) does not support cold reboot yet
                 reboot(duthost, localhost)
             else:
-                ptf_runner(ptfhost,
-                        "ptftests",
-                        "advanced-reboot.ReloadTest",
-                        platform_dir="ptftests",
-                        params=test_params,
-                        platform="remote",
-                        qlen=10000,
-                        log_file=log_file)
+                if test_params['reboot_type'] == reboot_ctrl_dict.get(REBOOT_TYPE_SOFT).get("command"):
+                    # advance-reboot test (on ptf) does not support SOFT reboot yet
+                    reboot(duthost, localhost, REBOOT_TYPE_SOFT)
```

Comment:
We should note that this will lead us to soft-upgrades being tested without any I/O.

Also, if advance-reboot does not support SOFT reboot, then how was `test_advance_reboot::test_soft_reboot` tested? That test also uses advanced-reboot scripts.

Reply:
I removed soft-reboot advanced test since we don't need the io test for soft-reboot case.

Reply:
Minor comment - the if block can be converted as `elif` instead.